test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,494 @@
+
+      owner = accounts[4];
+      worker = accounts[8];
+
+      mockOrganization = await MockOrganization.new(owner, worker);
+
+      let symbol = "OST",
+        name = "Simple Token",
+        decimals = 18;
+
+      utilityToken = await Token.new(
+        accounts[3],
+        symbol,
+        name,
+        decimals,
+        owner,
+        {from: accounts[0]},
+      );
+
+      coreAddress = accounts[3];
+      burnerAddress = accounts[6];
+      bountyAmount = new BN(100);
+      dummyStateRootProvider = accounts[2];
+
+      eip20Gateway = await EIP20Gateway.new(
+        mockToken.address,
+        baseToken.address,
+        dummyStateRootProvider,
+        bountyAmount,
+        mockOrganization.address,
+        burnerAddress,
+      );
+
+      await eip20Gateway.activateGateway(TestDataJSON.contracts.coGateway, {from: owner});
+      await eip20Gateway.setStorageRoot(
+        redeemRequest.blockNumber,
+        redeemRequest.storageRoot,
+      );
+
+    }
+  );
+
+  it('should pass when all the params are valid', async function () {
+
+      await confirmRedeemIntent(redeemRequest);
+
+    }
+  );
+
+  it('should return correct param', async function () {
+
+    let result = await eip20Gateway.confirmRedeemIntent.call(
+      redeemRequest.redeemer,
+      redeemRequest.nonce,
+      redeemRequest.beneficiary,
+      redeemRequest.amount,
+      redeemRequest.gasPrice,
+      redeemRequest.gasLimit,
+      redeemRequest.blockNumber,
+      redeemRequest.hashLock,
+      redeemRequest.storageProof,
+    );
+
+    messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+    assert.strictEqual(
+      result,
+      messageHash,
+      `Message hash from event must be equal to ${messageHash}.`,
+    );
+
+  });
+
+  it('should emit `RedeemIntentConfirmed` event.', async function () {
+
+    let tx = await eip20Gateway.confirmRedeemIntent(
+      redeemRequest.redeemer,
+      redeemRequest.nonce,
+      redeemRequest.beneficiary,
+      redeemRequest.amount,
+      redeemRequest.gasPrice,
+      redeemRequest.gasLimit,
+      redeemRequest.blockNumber,
+      redeemRequest.hashLock,
+      redeemRequest.storageProof,
+    );
+
+    let event = EventDecoder.getEvents(tx, eip20Gateway);
+
+    assert.isDefined(
+      event.RedeemIntentConfirmed,
+      'Event RedeemIntentConfirmed must be emitted.',
+    );
+
+    let eventData = event.RedeemIntentConfirmed;
+
+    let params = TestDataJSON.gateway.confirm_redeem_intent.params;
+
+    assert.strictEqual(
+      eventData._messageHash,
+      messageHash,
+      `Message hash from event must be equal to ${messageHash}.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemer,
+      params.redeemer,
+      `Redeemer address ${eventData._redeemer} from event must be equal to ${params.redeemer}.`,
+    );
+
+    let nonce = new BN(params.nonce, 16);
```

Comment:
We can directly use `redeemRequest.nonce` which is already in BN. And ideally, we should not use the stub data here as its already initialized in `redeemRequest`. Similarly for other test asserts.